server/repository/repository.go

```diff
@@ -100,8 +104,13 @@ func (s *Server) ListApps(ctx context.Context, q *RepoAppsQuery) (*RepoAppsRespo
 
 	items := make([]*AppInfo, 0)
 
-	for i := range ksonnetRes.Items {
-		items = append(items, &AppInfo{Type: string(repository.AppSourceKsonnet), Path: ksonnetRes.Items[i]})
+	for _, k := range ksonnetRes.Items {
+		for _, c := range componentRes.Items {
```

Comment:
Can we avoid a n^2 algorithm for identifying ksonnet apps? Maybe iterate ksonnet apps first, store them as candidates in a map, then validate that they have adjacent component directories while you iterate componentRes.

Reply:
O(2n) algorithm in place now.